xcdat/axis.py

```diff
@@ -124,6 +124,11 @@ def swap_lon_axis(
                 "orientations."
             )
 
+    # If the current axis orientation is the same as the desired axis
+    # orientation, `pass` and return the same dataset.
+    if new_lon.identical(lon):
+        return dataset
```

Comment:
The fix was to simply compare the original longitude coords with the swapped coords. If they are identical, return the original dataset. Otherwise, return the dataset with the swapped coords.